### PR TITLE
Delete plugin: unremovables and unmergeables

### PR DESCRIPTION
**Pull request by cammarosano**

### CODE REVIEW

File: addons/html_editor/static/src/editor/core/delete_plugin.js [hunk 1]
```diff
@@ -285,40 +301,51 @@ export class DeletePlugin extends Plugin {
...
+        // Should this change, the predicates must be adapted to take a Node
+        // instead of an Element as argument.
+        if (node.nodeType === Node.TEXT_NODE) {
+            return false;
+        }
+        if (node.nodeType !== Node.ELEMENT_NODE) {
+            return true;
+        }
+        return this.resources.unremovables.some((predicate) => predicate(node, root));
+    }
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/editor/core/delete_plugin.js:327]: Which type of node arrive in this condition ?

File: addons/html_editor/static/src/editor/core/delete_plugin.js [hunk 2]
```diff
@@ -285,40 +301,51 @@ export class DeletePlugin extends Plugin {
...
 
-        for (const { callback } of handleUnremovable) {
-            if (callback(node)) {
+    // Returns true if the entire subtree rooted at node was removed.
+    // Unremovable nodes take the place of removable ancestors.
+    removeNode(node) {
+        const root = node;
+        const remove = (node) => {
+            for (const child of [...node.childNodes]) {
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/editor/core/delete_plugin.js:334]: childNodes is iterable
```suggestion
            for (const child of node.childNodes) {
```

File: addons/html_editor/static/src/editor/core/delete_plugin.js (hunk 1 above)
Comment (cammarosano) [reply addons/html_editor/static/src/editor/core/delete_plugin.js:327]: Element

File: addons/html_editor/static/src/editor/core/delete_plugin.js (hunk 2 above)
Comment (cammarosano) [reply addons/html_editor/static/src/editor/core/delete_plugin.js:334]: `node.childNodes` returns a live NodeList. I need it frozen, as I'm making changes to its children.

File: addons/html_editor/static/src/editor/core/delete_plugin.js (hunk 1 above)
Comment (FrancoisGe) [reply addons/html_editor/static/src/editor/core/delete_plugin.js:327]: You right, I think the condition was inverse ^^

**Pull request merged**